addons/test_mass_mailing/tests/test_mailing_test.py

```diff
@@ -75,6 +75,39 @@ def test_mailing_test_button(self):
         with self.mock_mail_gateway(), self.assertRaises(Exception):
             mailing_test.send_mail_test()
 
+    @users('user_marketing')
+    @mute_logger('odoo.addons.mail.models.mail_render_mixin')
+    def test_mailing_test_button_links(self):
+        mailing = self.test_mailing_bl.with_env(self.env)
+        mailing_test = self.env['mailing.mailing.test'].create({
+            'email_to': 'test@test.com',
+            'mass_mailing_id': mailing.id,
+        })
+        # Test if link snippets are correctly converted
+        mailing.write({
+            'subject': 'Subject {{ object.name }}',
+            'preview': 'Preview {{ object.name }}',
+            'body_html':
+                '''<p>
+                Hello
+                    <div class="o_snippet_view_in_browser o_mail_snippet_general pt16 pb16" style="text-align: center; padding-left: 15px; padding-right: 15px;">
+                        <a href="/view">
```

Comment:
Just to be sure, we could include a link to http://www.example.com/view and check it has not been replaced :)